activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToAPITaskUpdatedEventConverter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.runtime.api.event.impl;
+
+import java.util.Optional;
+
+import org.activiti.api.task.runtime.events.TaskCreatedEvent;
+import org.activiti.api.task.runtime.events.TaskUpdatedEvent;
+import org.activiti.engine.delegate.event.ActivitiEntityEvent;
+import org.activiti.runtime.api.model.impl.APITaskConverter;
+
+public class ToAPITaskUpdatedEventConverter implements EventConverter<TaskUpdatedEvent, ActivitiEntityEvent> {
+
+    private final APITaskConverter taskConverter;
+
+    public ToAPITaskUpdatedEventConverter(APITaskConverter taskConverter) {
+        this.taskConverter = taskConverter;
+    }
+
+    @Override
+    public Optional<TaskUpdatedEvent> from(ActivitiEntityEvent internalEvent) {
+        return Optional.of(new TaskUpdatedEventImpl(taskConverter.from((org.activiti.engine.task.Task) internalEvent.getEntity())));
```

Comment:
Here you need to check before if it's a Task update or an update of another entity. As you are using `ActivitiEventType.ENTITY_UPDATED` to register the listener it's listening for all the entity updates, not only for tasks. See https://github.com/Activiti/Activiti/blob/develop/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/ToTaskSuspendedConverter.java#L37.

Reply:
I missed that, makes total sense to have this check.